### ISSUE TITLE
PLR1701 fix is not safe (especially together with SIM114)

### ISSUE THREAD

**Issue by simonpercivall** [astral-sh/ruff#11616]
`PLR1701` is mentioned in passing in https://github.com/astral-sh/ruff/issues/10245.

`PLR1701` will merge repeated `isinstance` calls and also rewrite literal tuples into unions for modern Python versions. But an `isinstance` check with a variable bound `tuple` will _also_ be merged into the union, and that will cause a `TypeError` at runtime.

This especially interacts badly with `SIM114` where you _could_ have defined two `if` arms, one with a variable bound `tuple`, and one with a literal `tuple`, and it'll get merged and wrongly rewritten as a union. Example below.

Ruff 0.45

`pyproject.toml`:
```toml
[tool.ruff]
target-version = "py311"
```

`ruff check --select SIM114,PLR1701 --fix`
```python
TYPES = (dict, list)

if isinstance(None, TYPES):
    pass
elif isinstance(None, set | float):
    pass
```

actual
```python
TYPES = (dict, list)

if isinstance(None, TYPES | set | float):
    pass

# Traceback (most recent call last):
#   File "PLR1701.py", line 3, in <module>
#     if isinstance(None, TYPES | set | float):
#                         ~~~~~~^~~~~
# TypeError: unsupported operand type(s) for |: 'tuple' and 'type'
```

**Comment by charliermarsh** [comment]
Thanks - we should mark it as unsafe. We can improve the rule in the future when we support type inference.
